src/utils/parse-arguments-into-options.ts

```diff
@@ -92,11 +44,12 @@ export async function parseArgumentsIntoOptions(
   // ToDo. Allow multiple
   const extension = extensionName ? await validateExternalExtension(extensionName, dev) : null;
 
-  if (!dev && extension && !CURATED_EXTENSIONS[args["--extension"] as string]) {
+  // if dev mode, extension would be a string
+  if (typeof extension === "object" && !extension?.isTrusted) {
```

Comment:
Not sure if the `=== "object"` thing is clearer than before.

The problem with the previous version is that TS doesn't infer that when !dev && extension => extension is an object. So you have to explicitly annotate the type.

Reply:
The comment makes it super clear 🙌

very very nitpick, but I think maybe we keep the condition as `if (extension && typeof extension === "object" && !extension.isTrusted)` so that we don't have to use `?.` even inside if condition.  

But yeah even current version look clean so feel free to ignore

Reply:
Like!

8f9616ec83654609791d889fa1885b987df42b04